x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -91,7 +99,7 @@ public LogicalPlan optimize(LogicalPlan verified) {
         Batch label = new Batch("Set as Optimized", Limiter.ONCE,
                 new SetAsOptimized());
 
-        return Arrays.asList(substitutions, operators, ordering, local, label);
+        return Arrays.asList(substitutions, operators, constraints, operators, ordering, local, label);
```

Comment:
Yes, it is because the operators need to be optimized before and after the new filters are added otherwise they'll be missed.

For example the filter `where a + 1 > 2 + 3 or true` will be skipped due to `or`. Further more there's folding and potentially some range combinations that might apply.
Likewise, after the new constraint was added, the filters need to be combined and potentially the resulting condition improved:

`where a > 1`
`where a == 0`

These two filters would be combined into one then another rule would pick it up and make the filter false which would result in the sequence being skipped.